image-loader/app/lib/imaging/FileMetadataReader.scala

```diff
@@ -125,17 +127,28 @@ object FileMetadataReader {
       }
     }
 
-  def getColorModelInformation(metadata: Metadata): Map[String, String] = {
+  def getColorModelInformation(image: File, metadata: Metadata): Future[Map[String, String]] = {
 
-      val pngDir = metadata.getFirstDirectoryOfType(classOf[PngDirectory])
+    val source = addImage(image)
+
+    val formatter = format(source)("%A")
```

Comment:
Turns out this just detects if the transparency is _supported_, not if it there, actually. Wholly opaque images like 1e364185d42b0663fe77c9b9b0128124ea712188 or 5c6cc57cfaf8392a631d5541b133d9774a441ba1 still slip through :(

If we are using ImageMagick, we should do [this](https://www.imagemagick.org/discourse-server/viewtopic.php?t=18596#p71706). If GraphicsMagick - [that](http://stackoverflow.com/a/39912666).

Reply:
Another possibly relevant bit https://github.com/guardian/grid/blob/master/cropper/app/lib/Crops.scala#L101

Reply:
We may also want to try
`gm -identify -format %r`
whenever it contains `Matte`, there is transparency. Tests needed, though.